static/js/upload.js

```diff
@@ -279,9 +277,7 @@ export function setup_upload(config) {
 
         const has_errors = get_item("send_status", config).hasClass("alert-error");
         if (!uploads_in_progress && !has_errors) {
-            setTimeout(() => {
-                hide_upload_status(config);
-            }, 500);
+            hide_upload_status(config);
```

Comment:
500ms seems excessive, but if you remove this timeout entirely, then users will never see the upload bar hit 100% -- I just tested with a 2MB file with no throttling on localhost, and with this PR, one sees the upload bar zoom along and then at like 60% it just disappears and the file is there. Some playing with it is likely required, but my guess is like a 50-300ms delay here will feel substantially nicer. I suspect if you watch closely with a major app like gmail, they have a similar delay before hiding.

Reply:
Ah! Cool, I can see doing a 100ms delay or something, and also adding a comment above the timeout for why it's there :)

Reply:
I'll keep the timeout from clicking the x button immediate.

Reply:
Updated!

Reply:
Actually ended up doing this:

```diff
@@ -141,9 +141,7 @@ export function upload_files(uppy, config, files) {
         compose_ui.autosize_textarea(get_item("textarea", config));
         uppy.cancelAll();
         get_item("textarea", config).trigger("focus");
-        setTimeout(() => {
-            hide_upload_status(config);
-        }, 500);
+        hide_upload_status(config);
     });
 
     for (const file of files) {
@@ -204,9 +202,14 @@ export function setup_upload(config) {
         },
     });
 
-    uppy.use(ProgressBar, {
-        target: get_item("send_status_identifier", config),
-        hideAfterFinish: false,
+    uppy.on("progress", (progress) => {
+        // When upload is complete, it resets to 0, which we don't want to do.
+        if (progress === 0) {
+            return;
+        }
+        $(`${get_item("send_status_identifier", config)} .moving_bar`).css({
+            width: `${progress}%`,
+        });
     });
 
     $("body").on("change", get_item("file_input_identifier", config), (event) => {
@@ -279,9 +282,11 @@ export function setup_upload(config) {
 
         const has_errors = get_item("send_status", config).hasClass("alert-error");
         if (!uploads_in_progress && !has_errors) {
+            // Hide upload status for 100ms after the 1s transition to 100%
+            // so that the user can see the progress bar at 100%.
             setTimeout(() => {
                 hide_upload_status(config);
-            }, 500);
+            }, 1100);
         }
     });
```

Reply:
This seems to work pretty well playing with it in practice!